dropshot/examples/https.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020 Oxide Computer Company
+
+/*!
+ * Example use of Dropshot with TLS enabled
+ */
+
+use dropshot::endpoint;
+use dropshot::ApiDescription;
+use dropshot::ConfigDropshot;
+use dropshot::ConfigLogging;
+use dropshot::ConfigLoggingLevel;
+use dropshot::HttpError;
+use dropshot::HttpResponseOk;
+use dropshot::HttpResponseUpdatedNoContent;
+use dropshot::HttpServerStarter;
+use dropshot::RequestContext;
+use dropshot::TypedBody;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+use std::io::Write;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use tempfile::NamedTempFile;
+
+/*
+ * This function would not be used in a normal application. It is used to
+ * generate temporary keys and certificates for the purpose of this demo.
+ */
+fn generate_keys() -> Result<(NamedTempFile, NamedTempFile), String> {
+    let keypair =
+        rcgen::generate_simple_self_signed(vec!["localhost".to_string()])
+            .map_err(|e| e.to_string())?;
+    let cert = keypair.serialize_pem().map_err(|e| e.to_string())?;
+    let priv_key = keypair.serialize_private_key_pem();
+
+    let mut cert_file =
+        NamedTempFile::new().map_err(|_| "failed to create cert_file")?;
+    cert_file
+        .write(cert.as_bytes())
+        .map_err(|_| "failed to write cert_file")?;
+    let mut key_file =
+        NamedTempFile::new().map_err(|_| "failed to create key_file")?;
+    key_file
+        .write(priv_key.as_bytes())
+        .map_err(|_| "failed to write key_file")?;
+    Ok((cert_file, key_file))
+}
+
+#[tokio::main]
+async fn main() -> Result<(), String> {
+    /*
+     * Begin by generating TLS certificates and keys. A normal application would
+     * just pass the paths to these via ConfigDropshot.
+     */
+    let (cert_file, key_file) = generate_keys()?;
```

Comment:
I love this example.  It concisely shows what you need to make a regular Dropshot server a TLS server.